util.py

```diff
@@ -12,6 +12,47 @@ class Day(IntEnum):
     SUNDAY = 6
 
 
+def returnNewDate(default_date, new_date):
+    #returns default day as an integer
+    #Monday is 0 sunday is 6
+    default_day = default_date.weekday()
+    today = datetime.datetime.now()
+    #if same day
+    if default_day == new_date:
+        return default_date
+    elif new_date < default_day:
+        difference = default_day - new_date
+        send_date = default_date - datetime.timedelta(days = difference)
+        #send date is in the past
+        if today > send_date:
+            send_date = send_date + datetime.timedelta(days = 7)
+            return send_date
+        else:
+        #send date is in the future
+            return send_date
+    else:
```

Comment:
I'm not too sure about the use of this function. From my understanding, get_next_datetime appears to already do this work for you, so I think this function is redundant - unless I'm misunderstanding your code, it's probably better to just call get_next_datetime rather than this in send.py.